File: week2_threes/gameOfThrees.py
#Game of threes using command-line args
import sys

def divide(x):
	print str(x)+"/3"
	x = x/3
	return x

def add(x):
	print str(x)+" + 1"
	x = x+1
	return x
	
def subtract(x):
	print str(x)+" - 1"
	x = x-1
	return x


x = int(sys.argv[1])
print x

while x > 1:
	if x%3 == 0:
		x = divide(x)
	elif (x+1)%3 == 0:
		x = add(x)
	elif (x-1)%3 == 0:
		x = subtract(x)

print x

File: week2_threes/GameOfThrees.cpp
#include <iostream>

using namespace std;

bool is_divisible(int x, int n);

int main()
{
    int x;

    cout << "=== Game of Threes ===" << endl << endl;

    // Ask user to enter number
    cout << "Enter number: ";
    cin >> x;

    do
        {
          // If number is divisible by 3, divide it by 3
          if (is_divisible(x, 3))
                x = x / 3;

          // Otherwise
          else
            {
              // If (number + 1) is divisible by 3, divide it
              if(is_divisible(x + 1, 3))
                  x = (x + 1)/3;

              // Or if (number - 1) is divisible by 3, divide it
              if (is_divisible(x - 1, 3))
                  x = (x - 1)/3;

            }

          // Output number
          cout << x << endl;
        }

    // And keep doing this while number is not 1
    while(x != 1);


    return 0;
}

bool is_divisible(int x, int n)
    {
      if(x % n == 0)
        return true;
      else
        return false;
    }
